### PR TITLE
DIM-6473: fix message for untransferable Postmaster items

### PR DESCRIPTION
**Pull request by goomba5**

### CODE REVIEW

File: src/app/inventory/DraggableInventoryItem.tsx [hunk 1]
```diff
@@ -110,7 +114,14 @@ function DraggableInventoryItem({
   const [touchActive, setTouchActive] = useState(false);
   const longPressed = useRef<boolean>(false);
   const timer = useRef<number>(0);
-  const isNotTransferable = item.notransfer && (!item.isEngram || !item.canPullFromPostmaster);
+  const stores = useSelector(sortedStoresSelector);
```
Comment (bhollis) [inline src/app/inventory/DraggableInventoryItem.tsx:117]: This makes sense, but we shouldn't add redux subscriptions to this component - it'd slow down DIM. We may have to find another way to do this that doesn't require looking at stores.

File: src/app/item-popup/ItemPopupBody.tsx [hunk 2]
```diff
@@ -27,10 +31,17 @@ export default function ItemPopupBody({
   onTabChanged(tab: ItemPopupTab): void;
 }) {
   const failureStrings = Array.from(extraInfo?.failureStrings || []);
-  const isNotTransferable = item.notransfer && (!item.isEngram || !item.canPullFromPostmaster);
   const isEngramCollected = item.location.hash === BucketHashes.Engrams;
+  const stores = useSelector(sortedStoresSelector);
```
Comment (bhollis) [inline src/app/item-popup/ItemPopupBody.tsx:35]: No need to use the sorted stores selector, can use regular stores

File: src/app/item-popup/ItemPopupBody.tsx [hunk 3]
```diff
@@ -27,10 +31,17 @@ export default function ItemPopupBody({
...
+  const store = getStore(stores, item.owner);
 
-  if ((isNotTransferable && item.location.inPostmaster) || isEngramCollected) {
+  if (!store) {
```
Comment (bhollis) [inline src/app/item-popup/ItemPopupBody.tsx:38]: Not all items are owned by a character - this will likely break vendor items.

**Review by bhollis — CHANGES_REQUESTED** [review]

**Comment by bhollis** [comment]
Some things to consider:

1. The original logic for showing the banner was probably OK, so we can go back to that. 
2. If you need Redux state for the item, ConnectedInventoryItem is the one place we have a subscription. 
3. We need to make sure the popup still works on the Vendors screen and the Progress screen. 

Since DIM automatically releases to production on Sunday afternoons, I'll probably revery your original change sometime between now and then to prevent it from going out, then we can take another pass at it.

**Comment by goomba5** [comment]
I appreciate the feedback and honesty. I'll keep an eye out for another ticket to work on.

**Pull request closed without merging**